core/block/editor/smartblock/smartblock.go

```diff
@@ -1037,6 +1041,7 @@ func (sb *smartBlock) StateRebuild(d state.Doc) (err error) {
 	if sb.IsDeleted() {
 		return ErrIsDeleted
 	}
+	sb.updateRestrictions()
```

Comment:
I tried but injectDerivedDetails works with passed state instead of sb.Doc
So need to rewrite RestrictionHolder to be compatible with the Doc. It still sounds like a good idea, though